### ISSUE TITLE
📝 In Vue, Biome does not recognize `lang=ts`, and fails to flag differing `indentStyle`

### ISSUE THREAD

**Issue by simnalamburt** [biomejs/biome#5513]
### Environment information

```bash
CLI:
  Version:                      2.0.0-beta.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v23.10.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.7.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /Users/simnalamburt/personal/biome-issue/biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Formatter:
  Format with errors:           unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Attribute position:           unset
  Bracket spacing:              unset
  Includes:                     unset

JavaScript Formatter:
  Enabled:                      unset
  JSX quote style:              unset
  Quote properties:             unset
  Trailing commas:              unset
  Semicolons:                   unset
  Arrow parentheses:            unset
  Bracket spacing:              unset
  Bracket same line:            unset
  Quote style:                  unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Attribute position:           unset

JSON Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Trailing Commas:              unset
  Expand lists:                 unset

CSS Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Quote style:                  unset

GraphQL Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Bracket spacing:              unset
  Quote style:                  unset

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
	"files": {
		"includes": ["**", "!**/dist"]
	}
}
```

### Playground link

https://github.com/simnalamburt/biome-issue-5513

The source code of `App.vue` is as follows. Although it uses `lang=ts` instead of `lang="ts"`, there’s no issue with the build. However, since the indentation uses spaces instead of tabs, Biome should be flagging this.

```vue
<script setup lang=ts>
// It's clearly a TypeScript block, but Biome fails to recognize it.
let num: number = 42;
</script>

<template>
  <h1>Hello, world!</h1>
  <p>num is {{ num }}</p>
</template>

<style scoped>
h1, p {
  font-family: sans-serif;
  color: #e3e3e3;
}
</style>
```

Instead, Biome fails to recognize the `lang=ts` attribute, so it doesn't know what language the code inside the `<script>` tag is written in and fails during parsing.

```console
$ pnpm biome format
/src/App.vue:2:54 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ type annotation are a TypeScript only feature. Convert your file to a TypeScript file or remove the syntax.

    1 │ <script setup lang=ts>
  > 2 │ // It's clearly a TypeScript block, but Biome fails to recognize it.
      │                                                      ^^^^^^^^
    3 │ let num: number = 42;
    4 │ </script>

  ℹ TypeScript only syntax


/Users/simnalamburt/personal/biome-issue/src/App.vue format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Code formatting aborted due to parsing errors. To format code with errors, enable the 'formatter.formatWithErrors' option.


Checked 9 files in 7ms. No fixes applied.
Found 1 error.
format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

Even if you manually fix that part, Biome still does not flag the indentation differences.

```console
$ perl -pi -e 's/lang=ts/lang="ts"/g' src/App.vue
$ pnpm biome format
Checked 9 files in 6ms. No fixes applied.
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
That's expected, it's part of the caveats: https://biomejs.dev/internals/language-support/#html-super-languages-support
